### PR TITLE
Added note about application registration limitations

### PR DESCRIPTION
**Pull request by elegault**
Some application requirements that need Exchange Online permissions need to use a different application registration method

### CODE REVIEW

**Comment by jasonjoh** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/add-ins/authenticate-a-user-with-an-sso-token.md](https://github.com/elegault/outlook-dev-docs/blob/patch-2/docs/add-ins/authenticate-a-user-with-an-sso-token.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2017%5C12%5C1%5Cf49385f6-041e-c905-242e-e0025c631156%5CPullRequest%5C201712012104490550-129%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=aB%2F7yfSI%2FnW0Kt2txFjIDFIurXCFQrwTF%2FbHP64PcVo%3D&st=2017-12-01T21%3A00%3A32Z&se=2018-01-01T21%3A05%3A32Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by jasonjoh** [comment]
@elegault This note seems out of place. I follow that getting an OAuth token for EWS requires registration in the Azure portal. Are you swapping the SSO token for an EWS-compatible token in your backend using the on-behalf-of flow?

**Comment by elegault** [comment]
I actually haven't got to the point where I can get a working token of any kind using the Office-js-helpers library.  But my intent is to use Azure auth to get token for logging into Exchange via the EWS Managed API. When I first setup my application registration I stumbled upon this inability to add Exchange Online permissions, so I thought this would be an appropriate article to post this "gotcha".  Besides, I'm scrambling to contribute to this Hack-A-Doc session that Julien Stroheker from MS is having us MVPs do right now live at MS Canada HQ lol. See: https://github.com/julienstroheker/MS-Hack-A-Doc

**Comment by jasonjoh** [comment]
For EWS we'd recommend just using the `Office.context.mailbox.getCallbackTokenAsync` method to get a token directly (assuming you're doing this from an Outlook add-in). No need to register anything if you do it that way :).

**Comment by elegault** [comment]
I would but from what I understand that's a read-only token? My goal is to update an FAI item (as per my other edit), which I can do from the Managed API but for now I'm using a username and password to login to EWS until I figure out the OAuth approach.

**Comment by jasonjoh** [comment]
If you specify `ReadWriteMailbox` in the `Permissions` element in the manifest, you should get a read/write token. I believe that should allow you to do what you want.

**Comment by elegault** [comment]
I'm pretty sure I tried that token and it wouldn't authorize, but I'll try it again to make sure.

**Comment by CameronDwyer** [comment]
@elegault I'd be interested if the ReadWriteMailbox permission allowed you to make changes to a Mail item via EWS. I've tried this and could still only get read access.

**Comment by jasonjoh** [comment]
You can modify items if you have the ReadWriteMailbox permission. (See https://docs.microsoft.com/en-us/outlook/add-ins/understanding-outlook-add-in-permissions#readwritemailbox-permission). If you weren't able to I'd be interested in seeing your code :)

**Comment by elegault** [comment]
The real problem is the error I get trying to update an FAI item from the add-in: "An Office add-in cannot update this kind of item". Thus the need to use the Managed EWS

**Comment by jasonjoh** [comment]
Ok. Are you not able to update an FAI with EWS + callback token?

**Comment by elegault** [comment]
From what I recall I was greatly confused on how to get that token, and looked to the office-js-helpers library to help me authenticate but was having problems getting that to work. So I'm simply using the user's password in the back-end to login to EWS (gasp!) and use the Managed API. I just needed something to work and have spent far too much time on something that shouldn't be this difficult to do

**Comment by jasonjoh** [comment]
Thanks for the feedback. So yes, FAI is an issue here.

- The `makeEwsRequestAsync` method is blocked from updating FAI per note [here](https://docs.microsoft.com/en-us/outlook/add-ins/web-services#ews-operations-that-add-ins-support).:

    > Note FAI (Folder Associated Information) items cannot be updated (or created) from an add-in. These hidden messages are stored in a folder and are used to store a variety of settings and auxiliary data. Attempting to use the UpdateItem operation will throw an ErrorAccessDenied error: "Office extension is not allowed to update this type of item".

- The callback token you get from `getCallbackTokenAsync` is scoped to the currently selected item, so it won't help to access FAI (or other non-FAI items in the mailbox).

Basically there isn't a good story for updating FAI items from an add-in. I'm not sure what your scenario is, but if your doing this to store settings, the intent for add-ins was to use the built-in methods to store any config data (https://docs.microsoft.com/en-us/outlook/add-ins/metadata-for-an-outlook-add-in). Of course if you're doing this for other reasons this may not be helpful :).

Regarding this pull request, I'm going to close it. This is important information, but it's not applicable to the SSO document which is about getting silent access to Graph. I think this probably belongs with the EWS documentation though, so I'll work with those folks to make sure it gets incorporated.

**Comment by elegault** [comment]
LOL - dude, you're quoting my edit to the article as to why makeEwsRequestAsync can't be used! Lesson learned the hard way haha!  My add-in needs to update the OWA signature, currently only possible using EWS (SOAP or Managed) from anywhere BUT an add-in.
